File: test/Z.Data.Test/System.Data.SqlClient.SqlParameterCollection/SqlParameterCollection.AddRangeWithValue.cs
// Description: C# Extension Methods Library to enhances the .NET Framework by adding hundreds of new methods. It drastically increases developers productivity and code readability. Support C# and VB.NET
// Website & Documentation: https://github.com/zzzprojects/Z.ExtensionMethods
// Forum: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License: https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: http://www.zzzprojects.com/
// Copyright © ZZZ Projects Inc. 2014 - 2016. All rights reserved.
using System.Collections.Generic;
using System.Data.SqlClient;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Z.Data.Test
{
    [TestClass]
    public class System_Data_SqlClient_SqlParameterCollection_AddRangeWithValue
    {
        [TestMethod]
        public void AddRangeWithValue()
        {
            const string sql = @"SELECT 1 As IntColumn WHERE @Value1 = 1 AND @Value2 = 2";

            var dict = new Dictionary<string, object> {{"@Value1", 1}, {"@Value2", 2}};

            // Examples
            using (var conn = new SqlConnection(My.Config.ConnectionString.UnitTest.ConnectionString))
            {
                using (SqlCommand @this = conn.CreateCommand())
                {
                    conn.Open();
                    @this.CommandText = sql;
                    @this.Parameters.AddRangeWithValue(dict);
                    var result = @this.ExecuteScalarAs<int>();

                    // UnitTest
                    Assert.AreEqual(1, result);
                }
            }
        }
    }
}

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Data/System.Data.SqlClient.SqlParameterCollection/SqlParameterCollection.AddRangeWithValue.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.Collections.Generic
Imports System.Data.SqlClient

Public Module Extensions_703

	''' <summary>
	'''     A SqlParameterCollection extension method that adds a range with value to 'values'.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <param name="values">The values.</param>
	<System.Runtime.CompilerServices.Extension> _
	Public Sub AddRangeWithValue(this As SqlParameterCollection, values As Dictionary(Of String, Object))
        For Each keyValuePair As Object In values
            this.AddWithValue(keyValuePair.Key, keyValuePair.Value)
        Next
	End Sub
End Module


